### PR TITLE
Closes #34 function for growth parameters for height/length

### LINKED ISSUE #34: New function: derive_params_growth_height - for BY LENGTH/HEIGHT growth parameters
**New function:**
```
derive_params_growth_height(
  dataset, 
  sex, 
  age, 
  age_unit, 
  meta_criteria, 
  parameter, 
  measure,
  height_age,
  set_values_to_sds, 
  set_values_to_pctl)
```
 
**Arguments:**

- dataset: input dataset
- sex, age and age_unit would define the variables to use from input dataset: e.g. sex = SEX, age = AAGECUR, age_unit = AAGECURU (age and age_unit are only needed if height_age argument is not NULL)
- meta_criteria: metadata as defined in step 4 of MS Teams doc
- parameter: the input dataset PARAMCD for which we want growth derivations: e.g. parameter = exprs(BMI = VSTESTCD == "BMI"). Options: “WEIGHT”
- measure: the input dataset variable for which we find the length/height at each time point
- height_age: the age in days at which the measure switches from body length to height as per the study CRF guidelines. If only ever length or height is used then leave this NULL and just feed in only the corresponding by length or height metadata (instead of the combined version which has both)
- set_values_to_sds: variables to be set for zscore (SDS) parameter: e.g. set_values_to_sds(PARAMCD = “WGTHSDS”, PARAM = “Weight-for-length/height z-score”). If left blank then parameter not derived in output dataset.
- set_values_to_pctl: variables to be set for percentile parameter: e.g. set_values_to_pctl(PARAMCD = “WGTHPCTL”, PARAM = “Weight-for-length/height Percentile”). If left blank then parameter not derived in output dataset.

**Checks:**

- dataset
  - The variable for sex is in input dataset 
  - The variables for age and age_unit are in input dataset if height_age is not NULL – age then first needs to get converted to DAYS using
  - The PARAMCD in parameter is available in input dataset as well as the variable in measure
  - the unit associated with each parameter is as expected (weight in kg)
- metadata
  - has variables for MEASURE, SEX, HEIGHT_LENGTH, L, M, S 

**Derivations:**

image

**Example function call:** 
```
advs_wgt_height <- advs %>% 
  derive_params_growth_height(
    sex = SEX, 
    age = AAGECUR, 
    age_unit = AAGECURU, 
    meta_criteria = mymetadata,
    parameter = WGHT,
    measure = HGTTMP,
    height_age = 730,
    set_values_to_sds(
      PARAMCD = “WGTHSDS”,
      PARAM = “Weight-for-length/height Z-Score”
    ),
    set_values_to_pctl(
      PARAMCD = “WGTHPCTL”,
      PARAM = “Weight-for-length/height Percentile”
    )
  )
```

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_params_growth_height.R [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+                                        analysis_var,
+                                        set_values_to_sds = NULL,
+                                        set_values_to_pctl = NULL) {
+  # Apply assertions to each argument to ensure each object is appropriate class
+  sex <- assert_symbol(enexpr(sex))
+  height <- assert_symbol(enexpr(height))
+  height_unit <- assert_symbol(enexpr(height_unit))
+  analysis_var <- assert_symbol(enexpr(analysis_var))
+  assert_data_frame(dataset, required_vars = expr_c(sex, height, height_unit, analysis_var))
+  assert_data_frame(meta_criteria, required_vars = exprs(SEX, HEIGHT, HEIGHTU, L, M, S))
```
Comment (Lina2689) [inline R/derive_params_growth_height.R:216]: @zdz2101 Can you change the input variables names HEIGHT/HEIGHTU coming from meta_criteria? as it's confusing when length will get selected, indeed in the [ADVS peds template](https://phuseaccount.sharepoint.com/:w:/r/sites/Pharmaverse-admiral/_layouts/15/Doc.aspx?sourcedoc=%7B5F12E4C8-0DB7-487E-B155-7C28D81B54D0%7D&file=ADVS_peds_template_draft.docx&action=default&mobileredirect=true), it should renamed as HEIGHT_LENGTH/HEIGHT_LENGTHU.

File: R/derive_params_growth_height.R [hunk 2]
```diff
@@ -0,0 +1,284 @@
+#' Derive Anthropometric indicators (Z-Scores/Percentiles-for-Age) based on Standard Growth Charts
+#'
+#' Derive Anthropometric indicators (Z-Scores/Percentiles-for-Age) based on Standard Growth Charts
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:3]: In these titles it mentions age, but isn't it for height/length now?

File: R/derive_params_growth_height.R [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+#' for Weight/BMI/Head Circumference by Height/Length
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:4]: We only have metadata for Weight here, not BMI and Head circumference

File: R/derive_params_growth_height.R [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+#'
+#' @param parameter Anthropometric measurement parameter to calculate z-score or percentile
+#'
+#'   A condition is expected with the input dataset `VSTESTCD`/`PARAMCD`
+#'   for which we want growth derivations:
+#'
+#'   e.g. `parameter = VSTESTCD == "WEIGHT"`.
+#'
+#'   There is CDC/WHO metadata available for Height, Weight, BMI, and Head Circumference available
+#'   in the `admiralpeds` package.
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:50]: Weight only in this case, and WHO only too (not CDC)

File: R/derive_params_growth_height.R [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+#'   Expected values: 'cm'
+#'
+#' @param meta_criteria Metadata dataset
+#'
+#'   A metadata dataset with the following expected variables:
+#'   `HEIGHT_LENGTH`, `HEIGHT_LENGTHU`, `SEX`, `L`, `M`, `S`
+#'
+#'   The dataset can be derived from CDC/WHO or user-defined datasets.
+#'   The CDC/WHO growth chart metadata datasets are available in the package and will
+#'   require small modifications.
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:34]: WHO only metadata for this one

File: R/derive_params_growth_height.R [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+#'
+#' @param analysis_var Variable containing anthropometric measurement
+#'
+#' A numeric vector is expected, e.g. `AVAL`, `VSSTRESN`
+#'
+#' @param set_values_to_sds Variables to be set for Z-Scores
+#'
+#'  The specified variables are set to the specified values for the new
+#'  observations. For example,
+#'   `set_values_to_sds(exprs(PARAMCD = “BMIASDS”, PARAM = “BMI-for-height z-score”))`
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:60]: Use weight instead of BMI as the example

File: R/derive_params_growth_height.R [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+#'
+#' *Permitted Values*: List of variable-value pairs
+#'
+#'  If left as default value, `NULL`, then parameter not derived in output dataset
+#'
+#' @param set_values_to_pctl Variables to be set for Percentile
+#'
+#'  The specified variables are set to the specified values for the new
+#'  observations. For example,
+#'   `set_values_to_pctl(exprs(PARAMCD = “BMIAPCTL”, PARAM = “BMI-for-height percentile”))`
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:71]: Use weight instead of BMI as the example

File: R/derive_params_growth_height.R [hunk 8]
```diff
@@ -0,0 +1,284 @@
+#' Derive Anthropometric indicators (Z-Scores/Percentiles-for-Age) based on Standard Growth Charts
+#'
+#' Derive Anthropometric indicators (Z-Scores/Percentiles-for-Age) based on Standard Growth Charts
+#' for Weight/BMI/Head Circumference by Height/Length
+#'
+#' @param dataset Input dataset
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:6]: Missing parameters from the issue: 
  age, 
  age_unit, 
  height_age,
These are all needed as we need to know at which age to assume height instead of body length but they're optional as: If only ever length or height is used then leave this NULL and just feed in only the corresponding by length or height metadata (instead of the combined version which has both)

I agree measure argument in the issue not needed - as we can add the height temp var to the input dataset

File: R/derive_params_growth_height.R [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+#'
+#'   The variables specified in `sex`, `age`, `age_unit`, `parameter`, `analysis_var`
+#'   are expected to be in the dataset.
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:9]: And height and height_unit too for this one

File: R/derive_params_growth_height.R [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+#' @examples
+#' library(dplyr)
+#' library(lubridate)
+#' library(rlang)
+#' library(admiral)
+#'
+#' advs <- dm_peds %>%
+#'   select(USUBJID, BRTHDTC, SEX) %>%
+#'   right_join(., vs_peds, by = "USUBJID") %>%
+#'   filter(USUBJID != "PEDS-1010") %>%
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:96]: are you using the latest metadat here as we use the `USUBJID` from pharmaversesdtm now? not these PEDS specific numbers

File: R/derive_params_growth_height.R [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+#'   mutate(
+#'     VSDT = ymd(VSDTC),
+#'     BRTHDT = ymd(BRTHDTC)
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AGECUR_M,
+#'     new_var_unit = CURU_M,
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:103]: can we use same names variables as the template for consistency here, e.g. AAGECUR

File: R/derive_params_growth_height.R [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+#'     HEIGHT_LENGTH = Height,
+#'     HEIGHT_LENGTHU = height_unit
+#'   )
+#'
+#'
+#' derive_params_growth_height(
+#'   advs_under2,
+#'   sex = SEX,
+#'   height = HEIGHT,
+#'   height_unit = HEIGHTU,
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:170]: same comment above to match template - think it was HGTTMP from memory? which is better as it doesn't then confuse with the height PARAMCD in VS, as hard to follow otherwise that this one is a variable being input not a parameter

File: R/derive_params_growth_height.R [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+#'   meta_criteria = who_under2,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   analysis_var = VSSTRESN,
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTASDS",
+#'     PARAM = "Weight-for-age z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTAPCTL",
+#'     PARAM = "Weight-for-age percentile"
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:180]: the parameters above are for age but should be for height/length

File: R/derive_params_growth_height.R [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
+#'   meta_criteria = who_over2,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   analysis_var = VSSTRESN,
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTASDS",
+#'     PARAM = "Weight-for-age z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTAPCTL",
+#'     PARAM = "Weight-for-age percentile"
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:198]: same comment as above

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
@zdz2101 i focused my review on the documentation and trust @kabis-ops and others to check the function code. and we can add more unit tests later as per the plan for the other function.

File: R/derive_params_growth_height.R (hunk 8 above)
Comment (rossfarrugia) [reply R/derive_params_growth_height.R:6]: This is so that you use the right metadata from WHO depending on patient age and which way they were likely measuring (height or body length).  it'd be good in the examples to use height_age = 730.5 days i.e. 2 years (even as default from what David explained to us?). See https://github.com/pharmaverse/admiralpeds/blob/35_advs_vignette/vignettes/advs.Rmd from line 229 for further explanation.

File: R/derive_params_growth_height.R (hunk 5 above)
Comment (rossfarrugia) [reply R/derive_params_growth_height.R:34]: @zdz2101 this one wasn't updated

File: R/derive_params_growth_height.R [hunk 15]
```diff
@@ -57,7 +56,7 @@
 #'
 #'  The specified variables are set to the specified values for the new
 #'  observations. For example,
-#'   `set_values_to_sds(exprs(PARAMCD = “BMIASDS”, PARAM = “BMI-for-height z-score”))`
+#'   `set_values_to_sds(exprs(PARAMCD = “WTASDS”, PARAM = “Weight-for-height z-score”))`
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:59]: These examples should be H instead of A in the PARAMCDs - as the A stands for AGE, H for HEIGHT

File: R/derive_params_growth_height.R [hunk 16]
```diff
@@ -93,14 +92,13 @@
 #' advs <- dm_peds %>%
 #'   select(USUBJID, BRTHDTC, SEX) %>%
 #'   right_join(., vs_peds, by = "USUBJID") %>%
-#'   filter(USUBJID != "PEDS-1010") %>%
 #'   mutate(
 #'     VSDT = ymd(VSDTC),
 #'     BRTHDT = ymd(BRTHDTC)
 #'   ) %>%
 #'   derive_vars_duration(
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:99]: `trunc_out = FALSE` can be removed as its default anyway for this function - to simplify the example

File: R/derive_params_growth_height.R [hunk 17]
```diff
@@ -93,14 +92,13 @@
 #' advs <- dm_peds %>%
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:92]: can we add comments to this example please? as its quite a lot for users to follow with so much pre-processing before we even get to the example function call

File: R/derive_params_growth_height.R (hunk 8 above)
Comment (rossfarrugia) [reply R/derive_params_growth_height.R:6]: @zdz2101 i'm open to your approach - it does give the user complete control. we'll just need to well comment this to explain our approach in your roxygen2 function documentation example (e.g. at the end you should add a comment to explain that the 2 resulting dataframes would need to be set back together to get the complete ADVS for this parameter) and also we'll need to explain well in our template comments and our vignette.

@Fanny-Gautier @Lina2689 what do you think? as the template authors i would trust your advice here as you'll have a good read on what makes this least complex for users.

Comment (Fanny-Gautier) [reply R/derive_params_growth_height.R:6]: In the template we selected the records from the metadata where MEASURE ="LENGTH" for patients <730.5 days, and MEASURE="HEIGHT" for patients >=730.5 days. As Zelos mentioned, it will require additional variables to merge the right data depending on the age.
We also added message in the ADVS peds template for the same `message("To derive height/length parameters, below function needs to call separately for Height and Length based on the input data and current age of the patient, as it depends on your CRF guidelines.")`.
I think it is easier to split it because if the user has only HEIGHT then there is only one call, similarly for LENGTH. But if the user has both LENGTH and HEIGHT in the data, it will complexify the merge. I think the user has more flexibility while splitting the derivation.

**Review by Fanny-Gautier — COMMENTED** [review]
looks like I didn't push my comments... sorry for the delay

File: R/derive_params_growth_height.R [hunk 18]
```diff
@@ -0,0 +1,284 @@
...
+#'   rename(
+#'     HEIGHT = VSSTRESN,
+#'     HEIGHTU = VSSTRESU
+#'   )
+#'
+#' advs <- advs %>%
+#'   right_join(., heights, by = c("USUBJID", "VSDTC"))
+#'
+#' advs_under2 <- advs %>%
+#'   filter(AGECUR_M <= 2)
```
[the remark this answers is not in the document]
Comment (zdz2101) [reply R/derive_params_growth_height.R]: my mistake this was fixed

File: R/derive_params_growth_height.R [hunk 19]
```diff
@@ -0,0 +1,284 @@
...
+#'     )
+#'   )
+#' ) %>%
+#'   rename(
+#'     HEIGHT = Height,
+#'     HEIGHTU = height_unit
+#'   )
+#'
+#'
+#' derive_params_growth_height(
```
[the remark this answers is not in the document]
Comment (zdz2101) [reply R/derive_params_growth_height.R:164]: patient should be in there, at least showing on my end

File: R/derive_params_growth_height.R [hunk 20]
```diff
@@ -0,0 +1,284 @@
...
+#'   )
+#'
+#' advs <- advs %>%
+#'   right_join(., heights, by = c("USUBJID", "VSDTC"))
+#'
+#' advs_under2 <- advs %>%
+#'   filter(AGECUR_M <= 2)
+#'
+#' advs_over2 <- advs %>%
+#'   filter(AGECUR_M > 2)
```
[the remark this answers is not in the document]
Comment (zdz2101) [reply R/derive_params_growth_height.R]: see above

File: R/derive_params_growth_height.R [hunk 21]
```diff
@@ -1,7 +1,8 @@
-#' Derive Anthropometric indicators (Z-Scores/Percentiles-for-Height) based on Standard Growth Charts
+#' Derive Anthropometric indicators (Z-Scores/Percentiles-for-Height/Length)
+#' based on Standard Growth Charts
 #'
-#' Derive Anthropometric indicators (Z-Scores/Percentiles-for-Height) based on Standard Growth Charts
-#' for Weight by Height/Length
+#' Derive Anthropometric indicators (Z-Scores/Percentiles-for-Height/Lenght)
```
Comment (rossfarrugia) [inline R/derive_params_growth_height.R:4]:
```suggestion
#' Derive Anthropometric indicators (Z-Scores/Percentiles-for-Height/Length)
```

File: R/derive_params_growth_height.R [hunk 22]
```diff
@@ -0,0 +1,282 @@
...
+#'
+#'
+#' @family der_prm_bds_vs
+#'
+#' @keywords der_prm_bds_vs
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
```
Comment (Minlei0201) [inline R/derive_params_growth_height.R:88]: Is line 88-198 supposed to be part of the examples?

File: R/derive_params_growth_height.R [hunk 23]
```diff
@@ -0,0 +1,282 @@
...
+#'     PARAMCD = "WTHPCTL",
+#'     PARAM = "Weight-for-height percentile"
+#'   )
+#' )
+derive_params_growth_height <- function(dataset,
+                                        sex,
+                                        height,
+                                        height_unit,
+                                        meta_criteria,
+                                        parameter,
```
Comment (Lina2689) [inline R/derive_params_growth_height.R:204]: @zdz2101 Can you please add HEIGHT_LENGTH and HEIGHT_LENGTHU as an argument to the function call as the metadata is now split for HEIGHT and LENGTH and we are calling the function separately for each parameter.  cc @rossfarrugia

Comment (rossfarrugia) [reply R/derive_params_growth_height.R:204]: Adding reply from the template comment here:
sorry @Lina2689 @Fanny-Gautier at the team call today i realised i had incorrectly said above "you dont even need a HEIGHT_LENGTH variable" - i meant MEASURE here as we don't need any variable to say "Height" or "Length" if we no longer join these into one metadata file. We still should make the values and units consistent so that they work with the function from @zdz2101 (i.e. HEIGHT_LENGTH and HEIGHT_LENGTHU).

So I think the above can be updated on template side, rather than new function arguments.

File: R/derive_params_growth_height.R [hunk 24]
```diff
@@ -0,0 +1,282 @@
...
+#'
+#'   A numeric vector is expected. Note that this is the actual height at the current visit.
+#'
+#' @param height_unit Height/Length Unit
+#
+#'   A character vector is expected.
+#'
+#'   Expected values: 'cm'
+#'
+#' @param meta_criteria Metadata dataset
```
Comment (Minlei0201) [inline R/derive_params_growth_height.R:28]: The original WHO metadata table has height/length increment of 0.5, but the metadata in the admiralpeds packages has increment of 0.1. How was it extrapolated? Do we have any documentation on that?

File: R/derive_params_growth_height.R [hunk 25]
```diff
@@ -0,0 +1,282 @@
...
+#'   The WHO growth chart metadata datasets are available in the package and will
+#'   require small modifications.
+#'   * `HEIGHT_LENGTH` - Height/Length
+#'   * `HEIGHT_LENGTHU` - Height/Length Unit
+#'   * `SEX` - Sex
+#'   * `L` - Power in the Box-Cox transformation to normality
+#'   * `M` - Median
+#'   * `S` - Coefficient of variation
+#'
+#' @param parameter Anthropometric measurement parameter to calculate z-score or percentile
```
Comment (Minlei0201) [inline R/derive_params_growth_height.R:43]: Shall we include a reminder that the expected unit of weight is kg?

File: R/derive_params_growth_height.R [hunk 26]
```diff
@@ -0,0 +1,282 @@
...
+#'   Expected values: 'cm'
+#'
+#' @param meta_criteria Metadata dataset
+#'
+#'   A metadata dataset with the following expected variables:
+#'   `HEIGHT_LENGTH`, `HEIGHT_LENGTHU`, `SEX`, `L`, `M`, `S`
+#'
+#'   The dataset can be derived from WHO or user-defined datasets.
+#'   The WHO growth chart metadata datasets are available in the package and will
+#'   require small modifications.
```
Comment (Minlei0201) [inline R/derive_params_growth_height.R:35]: The original WHO metadata table has height/length increment of 0.5, but the metadata in the admiralpeds packages has increment of 0.1. How was it extrapolated? Do we have any documentation on that?

File: R/derive_params_growth_height.R [hunk 27]
```diff
@@ -0,0 +1,282 @@
...
+#'
+#'   Expected values: 'cm'
+#'
+#' @param meta_criteria Metadata dataset
+#'
+#'   A metadata dataset with the following expected variables:
+#'   `HEIGHT_LENGTH`, `HEIGHT_LENGTHU`, `SEX`, `L`, `M`, `S`
+#'
+#'   The dataset can be derived from WHO or user-defined datasets.
+#'   The WHO growth chart metadata datasets are available in the package and will
```
Comment (Minlei0201) [inline R/derive_params_growth_height.R:34]: Shall we list out the names of the metadata to be clear, so that the users do not need to search for the metadata(or say "see example for details")?

File: R/derive_params_growth_height.R [hunk 28]
```diff
@@ -0,0 +1,282 @@
...
+#'   analysis_var = VSSTRESN,
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTHSDS",
+#'     PARAM = "Weight-for-height z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTHPCTL",
+#'     PARAM = "Weight-for-height percentile"
+#'   )
+#' )
```
Comment (Minlei0201) [inline R/derive_params_growth_height.R:198]: Shall we include codes to bind rows of `advs_under2` and `advs_over2`?

File: R/derive_params_growth_height.R [hunk 29]
```diff
@@ -0,0 +1,282 @@
...
+#'   mutate(
+#'     VSDT = ymd(VSDTC),
+#'     BRTHDT = ymd(BRTHDTC)
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AAGECUR,
+#'     new_var_unit = AAGECURU,
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "months"
```
Comment (Minlei0201) [inline R/derive_params_growth_height.R:105]: In the ad_advs.R template (PR #32 ), the unit of age is days (with 730.5 days as cutoff). Shall we be consistent in the example here?

**Comment by Minlei0201** [comment]
Hi @zdz2101 , I've added a few comments on the PR (on the `derive_params_growth_height.R`). Feel free to take a look and let me know if you have any follow-up questions!

File: R/derive_params_growth_height.R (hunk 26 above)
Comment (zdz2101) [reply R/derive_params_growth_height.R:35]: https://www.who.int/tools/child-growth-standards/standards/weight-for-length-height

scroll to the bottom on the `expanded tables` section you will find the increments were by 0.1

File: R/derive_params_growth_height.R (hunk 27 above)
Comment (zdz2101) [reply R/derive_params_growth_height.R:34]: This is a good idea, but I think we'll need to draw up a bigger vignette/article for metadata creation/preprocessing anyway

File: R/derive_params_growth_height.R (hunk 28 above)
Comment (zdz2101) [reply R/derive_params_growth_height.R:198]: Makes sense just for the sake of completing the example

File: R/derive_params_growth_height.R (hunk 29 above)
Comment (zdz2101) [reply R/derive_params_growth_height.R:105]: Yes I will fix this

File: R/derive_params_growth_height.R [hunk 30]
```diff
@@ -0,0 +1,289 @@
...
+#'   height_unit = HGTTMPU,
+#'   meta_criteria = who_under2,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   analysis_var = VSSTRESN,
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTHSDS",
+#'     PARAM = "Weight-for-height z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTHPCTL",
```
Comment (Lina2689) [inline R/derive_params_growth_height.R:182]: @zdz2101 please modify the PARAMCD to W**G**THSDS/W**G**THPCTL to be consistent with the ADVS template. Same changes are required for `derive_params_by_age()` function as well.
image
image

File: R/derive_params_growth_height.R (hunk 23 above)
Comment (Fanny-Gautier) [reply R/derive_params_growth_height.R:211]: Updated in ADVS peds template directly. Thanks

File: tests/testthat/test-derive_params_growth_height.R [hunk 31]
```diff
@@ -0,0 +1,41 @@
+# derive_params_growth_heightlength ----
+
+## Test 1: derive_params_growth_heightlength works ----
```
Comment (Fanny-Gautier) [inline tests/testthat/test-derive_params_growth_height.R:3]: Some outliers data to be added

**Review by Fanny-Gautier — COMMENTED** [review]
Minor comments added

File: tests/testthat/test-derive_params_growth_height.R (hunk 31 above)
Comment (zdz2101) [reply tests/testthat/test-derive_params_growth_height.R:3]: Additional testing will be handled in #50 #51
